### PR TITLE
🔏 Add validation at session assets level which returns missing assets

### PR DESCRIPTION
**Pull request by rowanseymour**
Replaces #567.

### CODE REVIEW

File: flows/actions/set_contact_field.go [hunk 1]
```diff
@@ -64,9 +64,11 @@ func (a *SetContactFieldAction) Execute(run flows.FlowRun, step flows.Step, logM
 	fields := run.Session().Assets().Fields()
 	field := fields.Get(a.Field.Key)
 
-	newValue := run.Contact().Fields().Parse(run.Environment(), fields, field, rawValue)
+	if field != nil {
```
Comment (rowanseymour) [inline flows/actions/set_contact_field.go]: this was only the only place where an action would blow up on a missing asset

File: flows/definition/dependencies.go [hunk 2]
```diff
@@ -47,13 +47,8 @@ func (d *dependencies) refresh(sa flows.SessionAssets, missing assets.MissingCal
 	}
 	for i, ref := range d.Fields {
 		a := sa.Fields().Get(ref.Key)
-
 		if a == nil {
-			// TODO for now if a field reference came from an expression (i.e. no name), we don't blow up if it's missing
```
Comment (rowanseymour) [inline flows/definition/dependencies.go]: since caller will now get back list of missing assets, it's up to them how strictly to deal with contact fields

**Comment by nicpottier** [comment]
So to be clear events will only be created for assets that exist in the session is that right? (just need to think through whether that needs to change in mailroom)

**Comment by rowanseymour** [comment]
yeah - if a group/label/field doesn't exist at runtime, we're going to noop that

**Pull request merged**